labgrid/driver/usbloader.py

```diff
@@ -337,3 +337,67 @@ def load(self, filename=None, phase=None):
     @step()
     def execute(self):
         pass
+
+
+@target_factory.reg_driver
+@attr.s(eq=False)
+class SamsungUSBDriver(Driver, BootstrapProtocol):
+    bindings = {
+        'loader': {'SamsungUSBLoader', 'NetworkSamsungUSBLoader'},
+    }
+
+    bl1 = attr.ib(validator=attr.validators.instance_of(str))
+    bl1_loadaddr = attr.ib(validator=attr.validators.instance_of(int))
+    spl_loadaddr = attr.ib(validator=attr.validators.instance_of(int))
+    loadaddr = attr.ib(validator=attr.validators.instance_of(int))
+    image = attr.ib(default=None)
+
+    def __attrs_post_init__(self):
+        super().__attrs_post_init__()
+        # FIXME make sure we always have an environment or config
+        if self.target.env:
+            self.tool = self.target.env.config.get_tool('smdk-usbdl')
+        else:
+            self.tool = 'smdk-usbdl'
+
+    def on_activate(self):
+        pass
+
+    def on_deactivate(self):
+        pass
+
+    @Driver.check_active
+    @step(args=['filename'])
+    def load(self, filename=None, phase=None):
+        if filename is None and phase == 'bl1':
+            filename = self.bl1
+        if filename is None and self.image is not None:
+            filename = self.target.env.config.get_image_path(self.image)
+        mf = ManagedFile(filename, self.loader)
+        mf.sync_to_resource()
```

Comment:
Shouldn't bl1, spl and the main image be handle consistently, so with get_image_path() for defaults and a way to select the image key name in the driver arguments?

Reply:
This information is provided by the caller, since it differs depending on the board being used